src/models/room-state.ts

```diff
@@ -420,6 +438,22 @@ export class RoomState extends TypedEventEmitter<EmittedEvents, EventHandlerMap>
             }
 
             const lastStateEvent = this.getStateEventMatching(event);
+
+            // Safety measure to not update the room (and emit the update) with older state.
+            // The sync loop really should not send old events but it does very regularly.
+            // Logging on return in those two conditions results in a large amount of logging. (on startup and when running element)
+            const lastReplaceId = lastStateEvent?.event.unsigned?.replaces_state;
+            const lastId = lastStateEvent?.event.event_id;
+            const newReplaceId = event.event.unsigned?.replaces_state;
+            const newId = event.event.event_id;
+            if (this._isStartTimelineState) {
+                // Add an event to the start of the timeline. Its replace id not be the same as the one of the current/last start state event.
+                if (newReplaceId && newId && newReplaceId === lastId) return;
+            } else {
+                // Add an event to the end of the timeline. It should not be the same as the one replaced but the current/last end state event.
+                if (lastReplaceId && newId && lastReplaceId === newId) return;
```

Comment:
I think one of these checks is wrong, they both check `newId` as a truthy but neither checks `lastId`